internal/buildscripts/packaging/fpm/common.sh

```diff
@@ -43,6 +48,36 @@ create_user_group() {
         sudo useradd --system --user-group --no-create-home --shell /sbin/nologin $SERVICE_USER
 }
 
+download_smart_agent() {
+    local tag="$1"
+    local buildroot="$2"
+    local api_url=""
+    local dl_url=""
+
+    if [ "$tag" = "latest" ]; then
+        tag=$( curl -sL "$SMART_AGENT_RELEASE_URL/latest" | jq -r '.tag_name' )
+        if [ -z "$tag" ]; then
+            echo "Failed to get tag_name for latest release from $SMART_AGENT_RELEASE_URL/latest" >&2
+            exit 1
+        fi
+    fi
+
+    api_url="$SMART_AGENT_RELEASE_URL/tags/$tag"
+    dl_url="$( curl -sL "$api_url" | jq -r '.assets[] .browser_download_url' | grep "signalfx-agent-${tag#v}\.tar\.gz" )"
+    if [ -z "$dl_url" ]; then
+        echo "Failed to get the agent download url from $api_url" >&2
+        exit 1
+    fi
+
+    echo "Downloading $dl_url ..."
+    curl -sL "$dl_url" -o "$buildroot/signalfx-agent.tar.gz"
+
+    mkdir -p "$buildroot/usr/lib"
+    tar -xzf "$buildroot/signalfx-agent.tar.gz" -C "$buildroot/usr/lib"
+    mv "$buildroot/usr/lib/signalfx-agent" "$buildroot/$SMART_AGENT_INSTALL_PATH"
+    rm -f "$buildroot/signalfx-agent.tar.gz"
```

Comment:
If we remove the agent binary we should free up a fair amount of space.

Reply:
Wasn't sure if we needed it or not. I'll remove it.

Reply:
🤞